crates/cheatcodes/src/env.rs

```diff
@@ -175,7 +175,11 @@ impl Cheatcode for envOr_4Call {
 impl Cheatcode for envOr_5Call {
     fn apply(&self, _state: &mut Cheatcodes) -> Result {
         let Self { name, defaultValue } = self;
-        env_default(name, defaultValue, &DynSolType::String)
+        let value = match env::var(name) {
+            Ok(env_var) if !env_var.is_empty() => env_var,
+            _ => defaultValue.to_string(),
+        };
```

Comment:
why only here?

Reply:
no particular reason, thought we want only for strings but was wrong. I changed to ensure env var not empty in https://github.com/foundry-rs/foundry/pull/9706/commits/a68a47cf35cb770fd7de517a7c7f5abfde07bdd6 please check